python/ffsim/states/slater.py

```diff
@@ -0,0 +1,353 @@
+        A Numpy array with the proposed new set of positions of the electrons.
+    """
+
+    rng = np.random.default_rng(rng)
+
+    n_chains = previous_step.shape[0]
+    nelec = previous_step.shape[1]
+
+    positions = np.arange(norb)
+
+    new_state = np.zeros((n_chains, nelec), dtype=int)
+
+    for i in range(n_chains):
+        sample_particles_to_move = rng.choice(np.arange(n_particles_to_move) + 1, 1)
+
+        open_orbitals = np.setdiff1d(positions, previous_step[i], assume_unique=True)
+
+        electron_id = np.sort(
+            rng.choice(np.arange(nelec), sample_particles_to_move, replace=False)
+        )
+
+        new_positions = rng.choice(
+            open_orbitals, sample_particles_to_move, replace=False
+        )
+
+        new_state[i] = previous_step[i]
+        new_state[i, electron_id] = new_positions
+        new_state[i] = np.sort(new_state[i])
+
+    return new_state
+
+
+def _evaluate_logdeterminant_squared(
+    positions: np.ndarray, phi: np.ndarray
+) -> tuple[np.ndarray, np.ndarray]:
+    """
+    Given a batch of positions of electronic configurations, evaluate their
+    Slater determinant.
+
+    Args:
+        positions: A 2D Numpy array of electronic configurations. Each row represents
+            the position of the electrons for each configuration.
+        phi: A 2D (norb, nelec) Numpy array. The matrix must be column-wise
+            orthogonal. Each column represents an orbital.
+
+    Returns: Tuple. The first element is a Numpy array with the signs of the
+        determinants. The second element is a Numpy array with the logarithm
+        of the absolute value of the determinant.
+    """
+
+    det_matrices = phi[positions, :]
+
+    signs, logdet_amps = np.linalg.slogdet(det_matrices)
+
+    return signs, logdet_amps
+
+
+def _accept(
+    logdet_squared_new: tuple[np.ndarray, np.ndarray],
+    logdet_squared_old: tuple[np.ndarray, np.ndarray],
+    rng=None,
+) -> np.ndarray:
+    """
+    Metropolis-Hastings acceptance of the new state of the chain.
+
+    Args:
+        logdet_squared_new: Tuple. The first element of the tuple is a Numpy
+            array with the sign of the determinant for the proposed electronic
+            configurations for each chain. The second element of the tuple is a
+            Numpy array with the logarithm of the absolute value of the
+            determinant of the proposed configurations.
+        logdet_squared_old: Tuple. The first element of the tuple is a Numpy
+            array with the sign of the determinant for the previous electronic
+            configurations for each chain. The second element of the tuple is a
+            Numpy array with the logarithm of the absolute value of the
+            determinant of the previous configurations.
+
+    Returns:
+        A Numpy array of ``bool`` type that specifies if the move for each chain
+        is accepted (``True``) of rejected (``False``).
+    """
+
+    rng = np.random.default_rng(rng)
```

Comment:
Same comment as above about the rng. It should be initialized only once in the main function.